### PR TITLE
Add 100km targets to longruns 

### LINKED ISSUE #752: Add 100km targets to longruns
The 100km resolution target AMIP runs to 300d [build](https://buildkite.com/clima/climacoupler-longruns/builds/649) , so it should replace the 200km target in longruns. 

## Components
- [x] add new specification
- [x] re-specify albedo type in driver, also so the config print is correct 

## Current setup
- longrun to 250d [build](https://buildkite.com/clima/climacoupler-longruns/builds/656#018f1ad5-055b-4534-b67f-11b5075cd06a)
  - toa radiation is much closer winter polar regions always too warm

Screen Shot 2024-04-30 at 9 22 49 PM

- debug plots after 70d [build](https://buildkite.com/clima/climacoupler-longruns/builds/663)
  - check land albedo
    - land albedo file seems ok
    - but the regridded version seems wrong #764 

Screen Shot 2024-04-30 at 9 25 31 PM
 
DEBUG PLOTS:
1) all coupler fields
Screen Shot 2024-04-30 at 9 23 24 PM

2) selected atmos fields
Screen Shot 2024-04-30 at 11 15 47 PM

3) selected bucket fields
Screen Shot 2024-04-30 at 11 15 36 PM

4) selected sea ice fields
Screen Shot 2024-04-30 at 11 16 14 PM

5) SST data model fields
Screen Shot 2024-04-30 at 11 16 40 PM

### PR DESCRIPTION
**Pull request by LenkaNovak**
closes #752

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: experiments/AMIP/user_io/debug_plots.jl [hunk 1]
```diff
@@ -34,13 +34,17 @@ function debug(cs_fields::NamedTuple, dir, cs_fields_ref = nothing)
...
+        :F_turb_ρτyz,
         :P_liq,
+        :P_snow,
         :T_S,
         :ρ_sfc,
         :q_sfc,
         :beta,
         :z0b_S,
         :z0m_S,
+        :radiative_energy_flux_toa,
```
Comment (juliasloan25) [inline experiments/AMIP/user_io/debug_plots.jl:47]: Have you tested interactively that these new fields are plotted correctly?

File: experiments/AMIP/coupler_driver.jl [hunk 2]
```diff
@@ -77,6 +77,7 @@ include("components/ocean/eisenman_seaice.jl")
 ## helpers for user-specified IO
 include("user_io/user_diagnostics.jl")
 include("user_io/user_logging.jl")
+include("user_io/debug_plots.jl")
```
Comment (juliasloan25) [inline experiments/AMIP/coupler_driver.jl:80]: Do we want to include this in main, or just output debug plots for interactive runs like we do now?

File: experiments/AMIP/components/atmosphere/climaatmos.jl [hunk 3]
```diff
@@ -64,6 +63,22 @@ function atmos_init(::Type{FT}, atmos_config_dict::Dict) where {FT}
     return sim
 end
 
+# Extension of CA.set_surface_albedo! to set the surface albedo to 0.38 at the beginning of the simulation,
+# so the initial callback initialization doesn't lead to NaNs in the radiation model.
+function CA.set_surface_albedo!(Y, p, t, ::CA.CouplerAlbedo)
```
Comment (juliasloan25) [inline experiments/AMIP/components/atmosphere/climaatmos.jl:68]: Do we need to extend this for `ConstantAlbedo` and `RegressionFunctionAlbedo` too, or is that done in Atmos?

File: experiments/AMIP/components/atmosphere/climaatmos.jl [hunk 4]
```diff
@@ -64,6 +63,22 @@ function atmos_init(::Type{FT}, atmos_config_dict::Dict) where {FT}
...
+    if t == 0
+        FT = CC.Spaces.undertype(axes(Y.c))
```
Comment (juliasloan25) [inline experiments/AMIP/components/atmosphere/climaatmos.jl:70]: I think we can also use `FT = eltype(Y)`, but either works :)

File: config/longrun_configs/gpu_amip_target_topo_diagedmf.yml [hunk 5]
```diff
@@ -1,26 +1,30 @@
...
 mode_name: "amip"
 mono_surface: false
 netcdf_output_at_levels: true
+output_default_diagnostics: true
+rayleigh_sponge: true
 run_name: "gpu_amip_target_topo_diagedmf"
 start_date: "19790301"
 surface_setup: "PrescribedSurface"
-t_end: "90days"
+t_end: "250days"
```
Comment (juliasloan25) [inline config/longrun_configs/gpu_amip_target_topo_diagedmf.yml:25]: Do we want this to be 300days, or is 250 enough?

**Review by juliasloan25 — COMMENTED** [review]
Looks great, thank you Lenka!

File: experiments/AMIP/user_io/debug_plots.jl (hunk 1 above)
Comment (LenkaNovak) [reply experiments/AMIP/user_io/debug_plots.jl:47]: I did. It's at the bottom of [this](https://github.com/CliMA/ClimaCoupler.jl/issues/752) issue. :)

File: experiments/AMIP/components/atmosphere/climaatmos.jl (hunk 3 above)
Comment (LenkaNovak) [reply experiments/AMIP/components/atmosphere/climaatmos.jl:68]: `water_albedo_from_atmosphere!` below is basically doing the `RegressionFunctionAlbedo` but at a different time than in atmos. I can clarify this in the comment :)

File: experiments/AMIP/components/atmosphere/climaatmos.jl (hunk 4 above)
Comment (LenkaNovak) [reply experiments/AMIP/components/atmosphere/climaatmos.jl:70]: Good point!

File: config/longrun_configs/gpu_amip_target_topo_diagedmf.yml (hunk 5 above)
Comment (LenkaNovak) [reply config/longrun_configs/gpu_amip_target_topo_diagedmf.yml:25]: Yes, with the increased timestep this should be possible, even with the diagnostics switched on.

File: experiments/AMIP/coupler_driver.jl (hunk 2 above)
Comment (juliasloan25) [reply experiments/AMIP/coupler_driver.jl:80]: I think it's fine to output them! Just wanted to check that it was intentional to leave this in

**Review by juliasloan25 — APPROVED** [review]
Looks good after CI is sorted!

**Pull request merged**